libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -530,18 +534,15 @@ - (void)syncUpOneEntry:(SFSyncState*)sync
      * passed in tells us to leave the record alone under these
      * circumstances, we will do nothing and return here.
      */
-    if (mergeMode == SFSyncStateMergeModeLeaveIfChanged &&
-        (action == SFSyncUpTargetActionUpdate || action == SFSyncUpTargetActionDelete)) {
+    if (mergeMode == SFSyncStateMergeModeLeaveIfChanged && !locallyCreated) {
         // Need to check the modification date on the server, against the local date.
         __weak SFSmartSyncSyncManager *weakSelf = self;
         SFSyncUpRecordModificationResultBlock modificationBlock = ^(NSDate *localDate, NSDate *serverDate, NSError *error) {
             __strong SFSmartSyncSyncManager *strongSelf = weakSelf;
-            if (error) {
-                if (failBlock != NULL) {
-                    failBlock(error);
-                }
-            } else if ([localDate compare:serverDate] != NSOrderedAscending) {
-                // Local date is newer than or the same as the server date.
+            if (localDate == nil // We didn't capture the last modified date so we can't really enforce merge mode
```

Comment:
The behavior was different that smartsync on android.
We should not fail to sync up if the modification date on the server cannot be established.